### PR TITLE
Enable ssh forwarding when building a remote target

### PR DESCRIPTION
**Pull request by alexcb**
this fixes https://github.com/moby/buildkit/issues/2040 by enabling ssh forwarding when a remote address is given on the command line, this is a similar fix to https://github.com/docker/buildx/pull/581

previously when I ran:

    DOCKER_BUILDKIT=1 ./build/docker -D build -t testing git@github.ent.corp:organization/testrepository.git

I could replicate the issue and would get this error:

    failed to solve with frontend dockerfile.v0: failed to read dockerfile: failed to load cache key: rpc error: code = Unimplemented desc = unknown service moby.sshforward.v1.SSH

However now with this fix, I get a much more appropriate error showing that my git clone failed:

```
 => ERROR [internal] load git source git@github.ent.corp:organization/testrepository.git                                                                                                                                                                                0.1s
------                                                                                                                                                                                                                                                                       
 > [internal] load git source git@github.ent.corp:organization/testrepository.git:
#1 0.105 ssh: Could not resolve hostname github.ent.corp: Name or service not known
#1 0.106 fatal: Could not read from remote repository.
#1 0.106 
#1 0.106 Please make sure you have the correct access rights
#1 0.106 and the repository exists.
```

and when I point it at an actual private repo, it can authenticate and builds as expected.

```
$ DOCKER_BUILDKIT=1 ./build/docker -D build -t testing 'git@github.com:alexcb/private-repo-with-dockerfile.git#main'
DEBU[0000] serving grpc connection                      
[+] Building 2.6s (5/5) FINISHED                                                                                                                                                                                                                                             
 => [internal] load git source git@github.com:alexcb/private-repo-with-dockerfile.git#main                                                                                                                                                                              1.3s
 => [internal] load metadata for docker.io/library/alpine:latest                                                                                                                                                                                                        0.0s
 => CACHED [1/2] FROM docker.io/library/alpine:latest                                                                                                                                                                                                                   0.0s 
 => [2/2] RUN echo hello private world                                                                                                                                                                                                                                  1.2s 
 => exporting to image                                                                                                                                                                                                                                                  0.0s 
 => => exporting layers                                                                                                                                                                                                                                                 0.0s 
 => => writing image sha256:4fad0d52c1e9a24e4ce4710b589bd94636c52ce8e275461e963ec81a1fe7d8b6                                                                                                                                                                            0.0s 
 => => naming to docker.io/library/testing  
```

### CODE REVIEW

File: vendor.conf [hunk 1]
```diff
@@ -47,7 +47,7 @@ github.com/Microsoft/go-winio                       5b44b70ab3ab4d291a7c1d28afe7
 github.com/Microsoft/hcsshim                        5bc557dd210ff2caf615e6e22d398123de77fc11 # v0.8.9
 github.com/miekg/pkcs11                             210dc1e16747c5ba98a03bcbcf728c38086ea357 # v1.0.3
 github.com/mitchellh/mapstructure                   d16e9488127408e67948eb43b6d3fbb9f222da10 # v1.3.2
-github.com/moby/buildkit                            8142d66b5ebde79846b869fba30d9d30633e74aa # v0.8.1
+github.com/moby/buildkit                            d3e0a6f2d8c36dcd2d785ab436ead8bd2dc50f28 # needed for util/gitutils
```
Comment (alexcb) [inline vendor.conf:50]: Should a v0.8.3 be released and referenced here instead? If so can someone help out with that?

Comment (thaJeztah) [reply vendor.conf:50]: If this is a bug fix for v0.8.x, we need to make sure it's cherry-picked into the v0.8 branch in the buildkit repository; do you know which pull request contains the fix in the buildkit repository?

**Comment by thaJeztah** [comment]
@tonistiigi @crazy-max ptal

File: vendor.conf (hunk 1 above)
Comment (thaJeztah) [reply vendor.conf:50]: From a quick look on GitHub, it looks like both https://github.com/moby/buildkit/pull/1782 and https://github.com/moby/buildkit/pull/1802 (which was a follow up) should be in the v0.8.2 buildkit tag;

Screenshot 2021-04-20 at 22 35 23
Screenshot 2021-04-20 at 22 35 32

If those are the changes needed, then I would suggest changing to;

```suggestion
github.com/moby/buildkit                            9065b18ba4633c75862befca8188de4338d9f94a # v0.8.2
```

**Review by tonistiigi — APPROVED** [review]

**Comment by tonistiigi** [comment]
@alexcb Validation + DCO broken

**Comment by alexcb** [comment]
It looks like the buildkit dependency version change that @thaJeztah and I made yesterday broke it https://github.com/docker/cli/pull/3050#discussion_r617015874

I suspect we'll need to update the 0.8 branch in the end (to include my PRs up until d3e0a6f2d8c36dcd2d785ab436ead8bd2dc50f28) to make this PR work. 

Should I open another PR against https://github.com/moby/buildkit/tree/v0.8 with the required cherry-picks to make this work?

**Comment by tonistiigi** [comment]
If you need something from master branch that isn't in `v0.8` (and isn't a regression bugfix that should be picked anyway) then just vendor from the commit you need.

**Comment by alexcb** [comment]
> If you need something from master branch that isn't in `v0.8` (and isn't a regression bugfix that should be picked anyway) then just vendor from the commit you need.

Does this mean I can re-enable my change to `vendor.conf`, to set it back to what I initially had it in under `github.com/moby/buildkit                            d3e0a6f2d8c36dcd2d785ab436ead8bd2dc50f28 # needed for util/gitutils`?

**Comment by tonistiigi** [comment]
Yes, @thaJeztah said that the commits you needed should have already been in the release tag. If this is not true then vendor the commit you need.

**Comment by alexcb** [comment]
thanks for the help @tonistiigi , I rebased this PR off master and the tests are once again passing.

**Review by tonistiigi — APPROVED** [review]

**Comment by avroshk** [comment]
Would love to get this merged @alexcb!
Please let me know if I can help resolve the conflicts.

**Comment by felipematheus** [comment]
I'm looking forward for this fix, would solve a problem with my build flow.

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**